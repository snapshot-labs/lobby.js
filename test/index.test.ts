import {
  getSpaceDelegationsIn,
  getSpaceDelegationsOut,
  getSpaceDelegations,
  getScores
} from '../src';
import addresses from './addresses.json';

const space = 'cvx.eth';
const network = '1';
const snapshot = 14780000;
const strategies = [
  {
    name: 'erc20-balance-of',
    params: {
      symbol: 'CVX',
      address: '0x72a19342e8F1838460eBFCCEf09F6585e32db86E',
      decimals: 18
    }
  },
  {
    name: 'eth-balance',
    params: {}
  }
];

describe('', () => {
  it('getSpaceDelegationsIn()', async () => {
    const delegationsIn = await getSpaceDelegationsIn(space, addresses, network, snapshot);
    expect(delegationsIn).toMatchSnapshot();
  }, 30e3);

  it('getSpaceDelegationsOut()', async () => {
    const delegationsOut = await getSpaceDelegationsOut(space, addresses, network, snapshot);
    expect(delegationsOut).toMatchSnapshot();
  }, 30e3);

  it('getSpaceDelegations()', async () => {
    const delegations = await getSpaceDelegations(space, addresses, network, snapshot);
    expect(delegations).toMatchSnapshot();
  }, 30e3);

  it('getScores() with delegation', async () => {
    const scores = await getScores(space, strategies, network, addresses, snapshot, true);
    expect(scores).toMatchSnapshot();
  }, 30e3);

  it('getScores() without delegation', async () => {
    const scores = await getScores(space, strategies, network, addresses, snapshot, false);
    expect(scores).toMatchSnapshot();
  }, 30e3);
});
