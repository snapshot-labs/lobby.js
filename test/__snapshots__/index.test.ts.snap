// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` getScores() with delegation 1`] = `
Object {
  "cache": true,
  "scores": Array [
    Object {
      "0x131bd1A2827ccEb2945B2e3B91Ee1Bf736cCbf80": 870579.9855681316,
      "0x14F83fF95D4Ec5E8812DDf42DA1232b0ba1015e6": 1738498.8133150414,
      "0x3aFB0B4cA9aB60165E207CB14067B07A04114413": 14703.31834524322,
    },
    Object {
      "0x131bd1A2827ccEb2945B2e3B91Ee1Bf736cCbf80": 0.6896301866118845,
      "0x14F83fF95D4Ec5E8812DDf42DA1232b0ba1015e6": 571.458574832913,
      "0x34807e0C31596a5BB70752827A4b37aB442E4D51": 0.16318777353680158,
      "0x3aFB0B4cA9aB60165E207CB14067B07A04114413": 437.3817544258386,
      "0x4C7909d6F029b3a5798143C843F4f8e5341a3473": 8.730873540558534,
      "0x59EE84E74662E824E44835322bf6078f4f93dD37": 0.013593910656164032,
      "0x676Eb0dC5644dBCB2C37849aD76432A73af0B582": 288.33205306186983,
      "0x6d9ABa400a2a487a5Fb76c6d56518835553CD284": 1.2460073153573592,
      "0xdD36Aa96D9BD4b49DA6E6734fF18Cc69F90F9435": 0.630682939591771,
    },
  ],
  "state": "final",
}
`;

exports[` getScores() without delegation 1`] = `
Object {
  "cache": true,
  "scores": Array [
    Object {
      "0x005EF868DA069044C86A3c68f9Cb31EC241D01C5": 39.81464612850746,
      "0x02eB66171AA8F78f3e0E6020A816F30A17E006dB": 12612.425948215934,
      "0x3aFB0B4cA9aB60165E207CB14067B07A04114413": 10469.879094454582,
      "0x3f426A31a77AfD8c2590146b8994755b0fc71349": 650.1142974020579,
      "0x5275817b74021E97c980E95EdE6bbAc0D0d6f3a2": 1170.253465768886,
      "0x66Ac8485cd335089d4A25AAEA082fC964976619b": 625.3564055554721,
      "0xBE0FafF90e05e93fceABFd068a44594a7B739742": 14471.144905472234,
      "0xebB0b94b7dB3C40661a2d6b1937530d08Daa6b53": 25117.797164746873,
    },
    Object {
      "0x00015f4a9B778A81cde7A9e6330Bf0E2b46A882b": 0.013593910656164032,
      "0x0013B99e50372D609f1E1A8Af83544878E99FcF7": 0.045634643180086,
      "0x0037f3Deb586d1b34aBAAe92341F9Bb70527a4d4": 3.0450907357497443,
      "0x005EF868DA069044C86A3c68f9Cb31EC241D01C5": 0.04162168478160965,
      "0x00b90B1FEb765c335DA3DC93CF89b7fB52C41cC3": 0.6407396914986414,
      "0x02eB66171AA8F78f3e0E6020A816F30A17E006dB": 0.6773753133663646,
      "0x073aBA73177668Ba1c6661A4CCddbF77dfC8809a": 76.92877544396639,
      "0x131bd1A2827ccEb2945B2e3B91Ee1Bf736cCbf80": 0.012254873245519925,
      "0x14F83fF95D4Ec5E8812DDf42DA1232b0ba1015e6": 0.024466,
      "0x34807e0C31596a5BB70752827A4b37aB442E4D51": 0.1175531303567156,
      "0x3aFB0B4cA9aB60165E207CB14067B07A04114413": 0.8351478762935352,
      "0x3f426A31a77AfD8c2590146b8994755b0fc71349": 0.7559706121081305,
      "0x4C7909d6F029b3a5798143C843F4f8e5341a3473": 8.467345773634987,
      "0x4a029dF7b4860fEcD1865C2cb387B89a65979835": 0.051132461764538825,
      "0x5275817b74021E97c980E95EdE6bbAc0D0d6f3a2": 0.4696033364545636,
      "0x59EE84E74662E824E44835322bf6078f4f93dD37": 0.013653040650457972,
      "0x66Ac8485cd335089d4A25AAEA082fC964976619b": 0.09339550275442715,
      "0x676Eb0dC5644dBCB2C37849aD76432A73af0B582": 73.49929265154859,
      "0x6d9ABa400a2a487a5Fb76c6d56518835553CD284": 1.2460073153573592,
      "0xBE0FafF90e05e93fceABFd068a44594a7B739742": 0.4259555713475672,
      "0xdD36Aa96D9BD4b49DA6E6734fF18Cc69F90F9435": 0.630682939591771,
      "0xeF8305E140ac520225DAf050e2f71d5fBcC543e7": 3.199092870617761,
      "0xebB0b94b7dB3C40661a2d6b1937530d08Daa6b53": 4.816211040691979,
    },
  ],
  "state": "final",
}
`;

exports[` getSpaceDelegations() 1`] = `
Object {
  "0x00015f4a9B778A81cde7A9e6330Bf0E2b46A882b": Object {
    "in": Array [],
    "out": "0x59EE84E74662E824E44835322bf6078f4f93dD37",
  },
  "0x0013B99e50372D609f1E1A8Af83544878E99FcF7": Object {
    "in": Array [],
    "out": "0x34807e0C31596a5BB70752827A4b37aB442E4D51",
  },
  "0x0037f3Deb586d1b34aBAAe92341F9Bb70527a4d4": Object {
    "in": Array [],
    "out": "0x676Eb0dC5644dBCB2C37849aD76432A73af0B582",
  },
  "0x005EF868DA069044C86A3c68f9Cb31EC241D01C5": Object {
    "in": Array [],
    "out": "0x947B7742C403f20e5FaCcDAc5E092C943E7D0277",
  },
  "0x00b90B1FEb765c335DA3DC93CF89b7fB52C41cC3": Object {
    "in": Array [],
    "out": "0x947B7742C403f20e5FaCcDAc5E092C943E7D0277",
  },
  "0x02eB66171AA8F78f3e0E6020A816F30A17E006dB": Object {
    "in": Array [],
    "out": "0x131bd1A2827ccEb2945B2e3B91Ee1Bf736cCbf80",
  },
  "0x073aBA73177668Ba1c6661A4CCddbF77dfC8809a": Object {
    "in": Array [],
    "out": "0x14F83fF95D4Ec5E8812DDf42DA1232b0ba1015e6",
  },
  "0x131bd1A2827ccEb2945B2e3B91Ee1Bf736cCbf80": Object {
    "in": Array [
      "0x02eB66171AA8F78f3e0E6020A816F30A17E006dB",
      "0x2d643Df5De4e9Ba063760d475BEAa62821c71681",
      "0xdFC95aaf0a107DaAe2b350458DED4b7906E7f728",
    ],
    "out": null,
  },
  "0x14F83fF95D4Ec5E8812DDf42DA1232b0ba1015e6": Object {
    "in": Array [
      "0x073aBA73177668Ba1c6661A4CCddbF77dfC8809a",
      "0x0D996171E7883A286eF720030935f72D0Bac8219",
      "0x186E20ae3530520C9F3E6C46F2f5d1062b784761",
      "0x19BB34410c9a0f70E89d1b9ab054059Ed1cf5aa8",
      "0x1eccd61c9fa53a8D2e823A26cD72A7efD7D0E92e",
      "0x1f575DC287742BDe83a5D7156f420d671A7C9b67",
      "0x3406402caE1f745714a9aFFc45939854e7Fbe12B",
      "0x37729c5948bb726a505B9633cAab02521c67d89a",
      "0x3A2D016124DF975850b84A6A970ECd5725bF793D",
      "0x3ff634ce65cDb8CC0D569D6d1697c41aa666cEA9",
      "0x465b357Bbac5F6f3BC78669Db6980f9Eaa21D0C2",
      "0x53D8EDF6a54239eB785eC72213919Fb6b6B73598",
      "0x595238341089097155d46A79970c38099ee011bC",
      "0x5A27DBBfF05F36DC927137855E3381f0c20C1CDd",
      "0x604eB5D4126E3318EC27721Bd5059307684F5C89",
      "0x628E8Ece00424f6EC1EBE2CBA684319f18079Bd5",
      "0x83b9711b4a5eA2A89606B4f0E4B0A0295B180Ab1",
      "0x843604564aA6fB2a28C06E20B02C30483BC2BcdC",
      "0x87fB47c2B9EB41d362BAb44F5Ec81514b6b1de13",
      "0x898111d1F4eB55025D0036568212425EE2274082",
      "0xa8e75d99D8D207B2916987c45722baC63D24D79f",
      "0xaBe780bDc3872B4DEC16bD44B2bB999f9151aE28",
      "0xB1139E0d8AE27B8a6EE26EC5296B9b18414eD856",
      "0xCA1D1A11074b1d86aC3fBD44771C7043126839c9",
      "0xcEd29BA48490C51E4348e654C313AC97762beCCC",
      "0xd8827D60486E01a6c54F7fc2ddCD9527a26af8ba",
      "0xE3e202b2CAC2B3eC41219C69fE0D2352b19cef69",
      "0xF7434638a7d52E44859A55fCf7c13DC9ddC4B140",
      "0xFd8dc2E9c828f87336042764784C042e489DB521",
    ],
    "out": null,
  },
  "0x34807e0C31596a5BB70752827A4b37aB442E4D51": Object {
    "in": Array [
      "0x0013B99e50372D609f1E1A8Af83544878E99FcF7",
    ],
    "out": null,
  },
  "0x3aFB0B4cA9aB60165E207CB14067B07A04114413": Object {
    "in": Array [
      "0x3f426A31a77AfD8c2590146b8994755b0fc71349",
      "0x4a029dF7b4860fEcD1865C2cb387B89a65979835",
      "0x5668EAd1eDB8E2a4d724C8fb9cB5fFEabEB422dc",
      "0xBA2258096034783747c1c62881AE09e4f318099c",
    ],
    "out": null,
  },
  "0x3f426A31a77AfD8c2590146b8994755b0fc71349": Object {
    "in": Array [],
    "out": "0x3aFB0B4cA9aB60165E207CB14067B07A04114413",
  },
  "0x4C7909d6F029b3a5798143C843F4f8e5341a3473": Object {
    "in": Array [
      "0x47D5592606622D807F3B37E5e5C8268355E30925",
    ],
    "out": null,
  },
  "0x4a029dF7b4860fEcD1865C2cb387B89a65979835": Object {
    "in": Array [],
    "out": "0x3aFB0B4cA9aB60165E207CB14067B07A04114413",
  },
  "0x5275817b74021E97c980E95EdE6bbAc0D0d6f3a2": Object {
    "in": Array [],
    "out": "0xde1E6A7ED0ad3F61D531a8a78E83CcDdbd6E0c49",
  },
  "0x59EE84E74662E824E44835322bf6078f4f93dD37": Object {
    "in": Array [
      "0x00015f4a9B778A81cde7A9e6330Bf0E2b46A882b",
    ],
    "out": "0xE485895fE0F7427830786dBa7eFfD8eFb0541d36",
  },
  "0x66Ac8485cd335089d4A25AAEA082fC964976619b": Object {
    "in": Array [],
    "out": "0xde1E6A7ED0ad3F61D531a8a78E83CcDdbd6E0c49",
  },
  "0x676Eb0dC5644dBCB2C37849aD76432A73af0B582": Object {
    "in": Array [
      "0x0037f3Deb586d1b34aBAAe92341F9Bb70527a4d4",
      "0x1Fdf7423979b7B64805893ee87c79b14e670C7Cf",
      "0x530cA90AAeC2957c771C707171d69413D9d1E35b",
      "0x716fd7a421b7EEaC5D5C658ddC824656e3e0F526",
      "0x77FBd7A41cfAB09b273c12Ecf3D217688fade9c5",
      "0xed75DAd20bDd2f916A329E0A2b88C8DE0DD00a28",
      "0xf7315745dF36D9baa254c63AD26a6Ff51d8e8E74",
    ],
    "out": null,
  },
  "0x6d9ABa400a2a487a5Fb76c6d56518835553CD284": Object {
    "in": Array [
      "0x1D65d296b6Ad7Fdbf77D3C9CC44C9b2787f1A341",
    ],
    "out": null,
  },
  "0xBE0FafF90e05e93fceABFd068a44594a7B739742": Object {
    "in": Array [],
    "out": "0xde1E6A7ED0ad3F61D531a8a78E83CcDdbd6E0c49",
  },
  "0xdD36Aa96D9BD4b49DA6E6734fF18Cc69F90F9435": Object {
    "in": Array [],
    "out": null,
  },
  "0xeF8305E140ac520225DAf050e2f71d5fBcC543e7": Object {
    "in": Array [],
    "out": "0xde1E6A7ED0ad3F61D531a8a78E83CcDdbd6E0c49",
  },
  "0xebB0b94b7dB3C40661a2d6b1937530d08Daa6b53": Object {
    "in": Array [],
    "out": "0xde1E6A7ED0ad3F61D531a8a78E83CcDdbd6E0c49",
  },
}
`;

exports[` getSpaceDelegationsIn() 1`] = `
Object {
  "0x00015f4a9B778A81cde7A9e6330Bf0E2b46A882b": Array [],
  "0x0013B99e50372D609f1E1A8Af83544878E99FcF7": Array [],
  "0x0037f3Deb586d1b34aBAAe92341F9Bb70527a4d4": Array [],
  "0x005EF868DA069044C86A3c68f9Cb31EC241D01C5": Array [],
  "0x00b90B1FEb765c335DA3DC93CF89b7fB52C41cC3": Array [],
  "0x02eB66171AA8F78f3e0E6020A816F30A17E006dB": Array [],
  "0x073aBA73177668Ba1c6661A4CCddbF77dfC8809a": Array [],
  "0x131bd1A2827ccEb2945B2e3B91Ee1Bf736cCbf80": Array [
    "0x02eB66171AA8F78f3e0E6020A816F30A17E006dB",
    "0x2d643Df5De4e9Ba063760d475BEAa62821c71681",
    "0xdFC95aaf0a107DaAe2b350458DED4b7906E7f728",
  ],
  "0x14F83fF95D4Ec5E8812DDf42DA1232b0ba1015e6": Array [
    "0x073aBA73177668Ba1c6661A4CCddbF77dfC8809a",
    "0x0D996171E7883A286eF720030935f72D0Bac8219",
    "0x186E20ae3530520C9F3E6C46F2f5d1062b784761",
    "0x19BB34410c9a0f70E89d1b9ab054059Ed1cf5aa8",
    "0x1eccd61c9fa53a8D2e823A26cD72A7efD7D0E92e",
    "0x1f575DC287742BDe83a5D7156f420d671A7C9b67",
    "0x3406402caE1f745714a9aFFc45939854e7Fbe12B",
    "0x37729c5948bb726a505B9633cAab02521c67d89a",
    "0x3A2D016124DF975850b84A6A970ECd5725bF793D",
    "0x3ff634ce65cDb8CC0D569D6d1697c41aa666cEA9",
    "0x465b357Bbac5F6f3BC78669Db6980f9Eaa21D0C2",
    "0x53D8EDF6a54239eB785eC72213919Fb6b6B73598",
    "0x595238341089097155d46A79970c38099ee011bC",
    "0x5A27DBBfF05F36DC927137855E3381f0c20C1CDd",
    "0x604eB5D4126E3318EC27721Bd5059307684F5C89",
    "0x628E8Ece00424f6EC1EBE2CBA684319f18079Bd5",
    "0x83b9711b4a5eA2A89606B4f0E4B0A0295B180Ab1",
    "0x843604564aA6fB2a28C06E20B02C30483BC2BcdC",
    "0x87fB47c2B9EB41d362BAb44F5Ec81514b6b1de13",
    "0x898111d1F4eB55025D0036568212425EE2274082",
    "0xa8e75d99D8D207B2916987c45722baC63D24D79f",
    "0xaBe780bDc3872B4DEC16bD44B2bB999f9151aE28",
    "0xB1139E0d8AE27B8a6EE26EC5296B9b18414eD856",
    "0xCA1D1A11074b1d86aC3fBD44771C7043126839c9",
    "0xcEd29BA48490C51E4348e654C313AC97762beCCC",
    "0xd8827D60486E01a6c54F7fc2ddCD9527a26af8ba",
    "0xE3e202b2CAC2B3eC41219C69fE0D2352b19cef69",
    "0xF7434638a7d52E44859A55fCf7c13DC9ddC4B140",
    "0xFd8dc2E9c828f87336042764784C042e489DB521",
  ],
  "0x34807e0C31596a5BB70752827A4b37aB442E4D51": Array [
    "0x0013B99e50372D609f1E1A8Af83544878E99FcF7",
  ],
  "0x3aFB0B4cA9aB60165E207CB14067B07A04114413": Array [
    "0x3f426A31a77AfD8c2590146b8994755b0fc71349",
    "0x4a029dF7b4860fEcD1865C2cb387B89a65979835",
    "0x5668EAd1eDB8E2a4d724C8fb9cB5fFEabEB422dc",
    "0xBA2258096034783747c1c62881AE09e4f318099c",
  ],
  "0x3f426A31a77AfD8c2590146b8994755b0fc71349": Array [],
  "0x4C7909d6F029b3a5798143C843F4f8e5341a3473": Array [
    "0x47D5592606622D807F3B37E5e5C8268355E30925",
  ],
  "0x4a029dF7b4860fEcD1865C2cb387B89a65979835": Array [],
  "0x5275817b74021E97c980E95EdE6bbAc0D0d6f3a2": Array [],
  "0x59EE84E74662E824E44835322bf6078f4f93dD37": Array [
    "0x00015f4a9B778A81cde7A9e6330Bf0E2b46A882b",
  ],
  "0x66Ac8485cd335089d4A25AAEA082fC964976619b": Array [],
  "0x676Eb0dC5644dBCB2C37849aD76432A73af0B582": Array [
    "0x0037f3Deb586d1b34aBAAe92341F9Bb70527a4d4",
    "0x1Fdf7423979b7B64805893ee87c79b14e670C7Cf",
    "0x530cA90AAeC2957c771C707171d69413D9d1E35b",
    "0x716fd7a421b7EEaC5D5C658ddC824656e3e0F526",
    "0x77FBd7A41cfAB09b273c12Ecf3D217688fade9c5",
    "0xed75DAd20bDd2f916A329E0A2b88C8DE0DD00a28",
    "0xf7315745dF36D9baa254c63AD26a6Ff51d8e8E74",
  ],
  "0x6d9ABa400a2a487a5Fb76c6d56518835553CD284": Array [
    "0x1D65d296b6Ad7Fdbf77D3C9CC44C9b2787f1A341",
  ],
  "0xBE0FafF90e05e93fceABFd068a44594a7B739742": Array [],
  "0xdD36Aa96D9BD4b49DA6E6734fF18Cc69F90F9435": Array [],
  "0xeF8305E140ac520225DAf050e2f71d5fBcC543e7": Array [],
  "0xebB0b94b7dB3C40661a2d6b1937530d08Daa6b53": Array [],
}
`;

exports[` getSpaceDelegationsOut() 1`] = `
Object {
  "0x00015f4a9B778A81cde7A9e6330Bf0E2b46A882b": "0x59EE84E74662E824E44835322bf6078f4f93dD37",
  "0x0013B99e50372D609f1E1A8Af83544878E99FcF7": "0x34807e0C31596a5BB70752827A4b37aB442E4D51",
  "0x0037f3Deb586d1b34aBAAe92341F9Bb70527a4d4": "0x676Eb0dC5644dBCB2C37849aD76432A73af0B582",
  "0x005EF868DA069044C86A3c68f9Cb31EC241D01C5": "0x947B7742C403f20e5FaCcDAc5E092C943E7D0277",
  "0x00b90B1FEb765c335DA3DC93CF89b7fB52C41cC3": "0x947B7742C403f20e5FaCcDAc5E092C943E7D0277",
  "0x02eB66171AA8F78f3e0E6020A816F30A17E006dB": "0x131bd1A2827ccEb2945B2e3B91Ee1Bf736cCbf80",
  "0x073aBA73177668Ba1c6661A4CCddbF77dfC8809a": "0x14F83fF95D4Ec5E8812DDf42DA1232b0ba1015e6",
  "0x131bd1A2827ccEb2945B2e3B91Ee1Bf736cCbf80": null,
  "0x14F83fF95D4Ec5E8812DDf42DA1232b0ba1015e6": null,
  "0x34807e0C31596a5BB70752827A4b37aB442E4D51": null,
  "0x3aFB0B4cA9aB60165E207CB14067B07A04114413": null,
  "0x3f426A31a77AfD8c2590146b8994755b0fc71349": "0x3aFB0B4cA9aB60165E207CB14067B07A04114413",
  "0x4C7909d6F029b3a5798143C843F4f8e5341a3473": null,
  "0x4a029dF7b4860fEcD1865C2cb387B89a65979835": "0x3aFB0B4cA9aB60165E207CB14067B07A04114413",
  "0x5275817b74021E97c980E95EdE6bbAc0D0d6f3a2": "0xde1E6A7ED0ad3F61D531a8a78E83CcDdbd6E0c49",
  "0x59EE84E74662E824E44835322bf6078f4f93dD37": "0xE485895fE0F7427830786dBa7eFfD8eFb0541d36",
  "0x66Ac8485cd335089d4A25AAEA082fC964976619b": "0xde1E6A7ED0ad3F61D531a8a78E83CcDdbd6E0c49",
  "0x676Eb0dC5644dBCB2C37849aD76432A73af0B582": null,
  "0x6d9ABa400a2a487a5Fb76c6d56518835553CD284": null,
  "0xBE0FafF90e05e93fceABFd068a44594a7B739742": "0xde1E6A7ED0ad3F61D531a8a78E83CcDdbd6E0c49",
  "0xdD36Aa96D9BD4b49DA6E6734fF18Cc69F90F9435": null,
  "0xeF8305E140ac520225DAf050e2f71d5fBcC543e7": "0xde1E6A7ED0ad3F61D531a8a78E83CcDdbd6E0c49",
  "0xebB0b94b7dB3C40661a2d6b1937530d08Daa6b53": "0xde1E6A7ED0ad3F61D531a8a78E83CcDdbd6E0c49",
}
`;
